src/easylink/runner.py

```diff
@@ -161,6 +163,32 @@ class FilteringOutput:
         def __init__(self, log_file_path: Path):
             self.log_file = open(log_file_path, "w")
             self.buffer = ""
+            self.last_output_time = time.time()
+            self.heartbeat_timer = None
+            self.dots_printed = False  # Track if we've printed progress dots
+            self._start_heartbeat()
+
+        def _start_heartbeat(self):
```

Comment:
Type hinting isn't failing mypy? I saw the same in the previous logging PR

Reply:
We turned mypy off for easylink a while back